src/components/va-text-input/test/va-text-input.e2e.ts

```diff
@@ -176,4 +176,48 @@ describe('va-text-input', () => {
       '(Max. 3 characters)',
     );
   });
+
+  it('allows manually setting the inputmode attribute', async () => {
+    const page = await newE2EPage();
+    await page.setContent('<va-text-input inputmode="numeric" />');
+
+    // Level-setting expectations
+    const inputEl = await page.find('va-text-input >>> input');
+    expect(inputEl.getAttribute('inputmode')).toBe('numeric');
+
+    // Test the functionality
+    await inputEl.press('2');
+    await inputEl.press('0');
+    expect(await inputEl.getProperty('value')).toBe('20');
```

Comment:
:thinking: I'm kind of torn on these tests. On the one hand, it's great to make sure it behaves as expected. I almost want you to add the failure mode to the tests as well. (Typing invalid data doesn't work.) I think instead, though, I'd prefer to leave this level of testing alone. That is, what we're doing here isn't testing that our web component behaves as expected, but that the underlying _browser_ behaves as expected on native elements (because we're interfacing directly with the `<input>` here).

How do you feel about not typing anything into the `<input>` and instead have the following tests?
1. Iterate over an allow-list for valid `type`s, expecting that the `type` attribute is set correctly for each
2. Check that an invalid `type` sets the `type` attribute on the `<input>` element to `text`
3. Iterate over an allow-list for valid `inputmode`s, expecting that the `inputmode` attribute is set correctly for each
4. Check that an invalid `inputmode` sets the `inputmode` attribute on the `<input>` element to whatever the default is (nothing?)

Reply:
> That is, what we're doing here isn't testing that our web component behaves as expected, but that the underlying browser behaves as expected on native elements

That makes sense, I'll update the tests :-)

Reply:
Changed to add #s 1-3, not doing #4 unless we decide to explicitly restrict allowed values, see the other comment. :-)